server/util.go

```diff
@@ -20,16 +20,27 @@ const (
 )
 
 // parseSize expects decimal positive numbers. We
-// return -1 to signal error
+// return -1 to signal error.
 func parseSize(d []byte) (n int) {
-	if len(d) == 0 {
+	var i, l int
+	var dec byte
+	l = len(d)
+	if l == 0 {
```

Comment:
nitpick: Could you replace var declarations above with this:
```
l := len(d)
if l == 0 {
  return -1
}
var (
  i int
  dec byte
)
```
(assuming that this does not affect performance)

Reply:
Thanks, updated and confirmed we also get a small performance boost:

```
benchmark                           old ns/op     new ns/op     delta
BenchmarkParseSize-4                8.81          5.73          -34.96%
Benchmark_____Pub0b_Payload-4       92.9          85.0          -8.50%
Benchmark_____Pub8b_Payload-4       96.1          85.4          -11.13%
Benchmark____Pub32b_Payload-4       106           99.3          -6.32%
Benchmark___Pub128B_Payload-4       131           122           -6.87%
Benchmark___Pub256B_Payload-4       152           146           -3.95%
Benchmark_____Pub1K_Payload-4       348           332           -4.60%
Benchmark_____Pub4K_Payload-4       1474          1428          -3.12%
Benchmark_____Pub8K_Payload-4       3330          3257          -2.19%

benchmark                           old MB/s     new MB/s     speedup
BenchmarkParseSize-4                113.53       174.60       1.54x
Benchmark_____Pub0b_Payload-4       118.45       129.37       1.09x
Benchmark_____Pub8b_Payload-4       197.74       222.43       1.12x
Benchmark____Pub32b_Payload-4       413.47       442.96       1.07x
Benchmark___Pub128B_Payload-4       1072.63      1154.81      1.08x
Benchmark___Pub256B_Payload-4       1765.96      1835.92      1.04x
Benchmark_____Pub1K_Payload-4       2979.37      3117.37      1.05x
Benchmark_____Pub4K_Payload-4       2788.11      2877.11      1.03x
Benchmark_____Pub8K_Payload-4       2464.19      2519.17      1.02x
```